.github/workflows/.check-majority-and-announce.yml

```
@@ -0,0 +1,219 @@
+name: Check SVS majority and Announce
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        required: true
+        type: string
+      info_url:
+        required: true
+        type: string
+      validator_operators_group_id:
+        required: false
+        type: string
+        default: "133138"  # Default group ID for validator operators
+    secrets:
+      SLACK_WEBHOOK_URL:
+        required: true
+      GROUPS_IO_USER:
+        required: true
+      GROUPS_IO_PASS:
+        required: true
+
+permissions: {}
+
+jobs:
+  check-and-announce:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # Need to write to announce-tracking branch
+      packages: write # Needed for Devnet only. Write to packages when copy images/charts from splice.
+    timeout-minutes: 20
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          path: main
+
+      - name: Checkout announce-tracking branch
+        uses: actions/checkout@v4
+        with:
+          ref: announce-tracking
+          path: tracking
+
+      - name: Set variables and Get GSF version and docs path
+        id: variables
+        env:
+          GSF_INFO_URL: ${{ inputs.info_url }}
+          GSF_ENV: ${{ inputs.environment }}
+        run: |
+          GSF_VERSION=$(curl -sSm5 "${GSF_INFO_URL}/info" | jq -r .sv.version)
+          GSF_DOCS=$(echo "${GSF_INFO_URL}" | sed 's/global.canton.network.//')
+          echo "GSF_INFO_URL=${GSF_INFO_URL}" >> $GITHUB_ENV
+          echo "GSF_ENV=${GSF_ENV}" >> $GITHUB_ENV
+          echo "GSF_VERSION=$GSF_VERSION" >> $GITHUB_ENV
+          echo "GSF_DOCS=$GSF_DOCS" >> $GITHUB_ENV
+
+      - name: Run SVS majority check
+        id: check-majority
+        env:
+          GSF_INFO_URL: ${{ env.GSF_INFO_URL }}
+          GSF_ENV: ${{ env.GSF_ENV }}
+          GITHUB_OUTPUT: $GITHUB_OUTPUT
+        run: |
+          chmod +x ./main/ci/check-svs-majority.sh
+          result=$(./main/ci/check-svs-majority.sh "${GSF_ENV}" "${GSF_INFO_URL}")
+          echo "$result"
+
+          {
+            echo '```'
+            echo "$result"
+            echo '```'
+          } >> "$GITHUB_STEP_SUMMARY"
+
+      - name: Check if version already announced
+        id: check-announced
+        env:
+          GSF_ENV: ${{ env.GSF_ENV }}
+          GSF_VERSION: ${{ env.GSF_VERSION }}
+        run: |
+          ENV_LOWER=$(echo "${GSF_ENV}" | tr '[:upper:]' '[:lower:]')
+          FILE="tracking/ci/announced-${ENV_LOWER}.txt"
+          if [ -f "$FILE" ] && grep -q "${GSF_VERSION}" "$FILE"; then
```

Add `-x` to match the whole line and `-F` to match fixed string instead of regex
```suggestion
          if [ -f "$FILE" ] && grep -qxF "${GSF_VERSION}" "$FILE"; then
```